src/test/java/org/logstash/beats/ServerTest.java

```diff
@@ -34,9 +42,43 @@ public void setUp() {
     @Test
     public void testServerShouldTerminateConnectionIdleForTooLong() throws InterruptedException {
         int inactivityTime = 3; // in seconds
+        int concurrentConnections = 10;
+
+        final Random random = new Random();
+
+        final CountDownLatch latch = new CountDownLatch(concurrentConnections);
 
         final Server server = new Server(host, randomPort, inactivityTime);
+        final AtomicBoolean otherCause = new AtomicBoolean(false);
+        server.setMessageListener(new MessageListener() {
+            public void onNewConnection(ChannelHandlerContext ctx) {
+                // Make sure connection is closed on exception too.
+                if (random.nextInt(10) < 1) {
```

Comment:
If the probability is high enough, all connection can get exception. In that case, no connection waits for defined timeout and assertion that checks it would fail.
We might want to create another test for error cases.